src/simulation.cpp

```diff
@@ -464,17 +463,26 @@ void finalize_batch()
   }
 
   // Write out surface source if requested.
-  if (settings::surf_source_write &&
-      simulation::current_batch == settings::n_batches) {
-    auto filename = settings::path_output + "surface_source";
-    auto surf_work_index =
-      mpi::calculate_parallel_index_vector(simulation::surf_source_bank.size());
-    gsl::span<SourceSite> surfbankspan(simulation::surf_source_bank.begin(),
-      simulation::surf_source_bank.size());
-    if (settings::surf_mcpl_write) {
-      write_mcpl_source_point(filename.c_str(), surfbankspan, surf_work_index);
-    } else {
-      write_source_point(filename.c_str(), surfbankspan, surf_work_index);
+  if (settings::surf_source_write) {
+    if (std::find(settings::surface_source_batch.begin(),
+          settings::surface_source_batch.end(),
+          simulation::current_batch) != settings::surface_source_batch.end()) {
```

Comment:
Ensure `container_util.h` is included in this file and you can use:

```suggestion
    if (contains(settings.surface_source_batch, simulation::current_batch)) {
```

Reply:
Perfect, it's already included.